v2/yarpcprotobuf/protoc-gen-yarpc-go/internal/generator/imports.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package generator
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Imports maps a set of import paths to unique aliases.
+type Imports map[string]string
+
+// Add adds the path to the imports map, initially using the base directory
+// as the package alias. If this alias is already in use, we continue to
+// prepend the remaining filepath elements until we have receive a unique
+// alias. If all of the path elements are exhausted, a '_' is continually used
+// until we create a unique alias.
+//
+//   imports := NewImports("json")
+//   imports.Add("encoding/json") -> "encodingjson"
+//   imports.Add("encodingjson")  -> "_encodingjson"
```

Comment:
Yea, that's fine with me. I'll update.

Reply:
Discussed offline - this actually adds a small amount of unnecessary complexity due to updating the number in place, as opposed to simply always prepending or appending. We'll leave this as is for now since it's unlikely we'll have collisions at all. And if we do, they should be easy to differentiate.